File: 11-pythonic-obj/private/Expose.java
import java.lang.reflect.Field;

public class Expose {

    public static void main(String[] args) {
        Confidential message = new Confidential("top secret text");
        Field secretField = null;
        try {
            secretField = Confidential.class.getDeclaredField("secret");
        }
        catch (NoSuchFieldException e) {
            System.err.println(e);
            System.exit(1);
        }
        secretField.setAccessible(true); // 破除访问限制！
        try {
            String wasHidden = (String) secretField.get(message);
            System.out.println("message.secret = " + wasHidden);
        }
        catch (IllegalAccessException e) { 
            // 调用 setAccessible(true) 之后不会发生
            System.err.println(e);
        }   
    }
}

File: 11-pythonic-obj/private/expose.py
#!/usr/bin/env jython
# 注意：截至 2020 年底，Jython 仍是 Python 2.7

import Confidential

message = Confidential('top secret text')
secret_field = Confidential.getDeclaredField('secret')
secret_field.setAccessible(True)  # 破除访问限制！
print 'message.secret =', secret_field.get(message)
